manifests/unit.pp

```diff
@@ -22,54 +22,86 @@
   $install_wantedby        = undef,
 ) {
 
-  validate_re($ensure, [ '^present$', '^absent$' ],
-    "systemd::unit::${name}::ensure is invalid and does not match the regex.")
+  include ::systemd
+
+  if is_string($ensure) {
```

Comment:
is it necessary to validate it's a string before the function validate_re? What is the error message of validate_re when it's not a string?

Reply:
Spec tests fails with "invalid" values because the module fails to compile correctly. This extra code is to avoid this situation.